pkg/codegen/convert/plugin_mapper.go

```diff
@@ -173,12 +173,87 @@ func NewPluginMapper(ws Workspace,
 	}, nil
 }
 
+// getMappingForPlugin calls GetMapping on the given plugin and returns it's result. This currently takes the
+// target provider name as an argument so that if we don't find the mapping for "terraform" we can try again
+// for "tf".
+func (l *pluginMapper) getMappingForPlugin(targetProvider string, pluginSpec mapperPluginSpec) ([]byte, string, error) {
+	providerPlugin, err := l.providerFactory(pluginSpec.name, &pluginSpec.version)
+	if err != nil {
+		// We should maybe be lenient here and ignore errors but for now assume it's better to fail out on
+		// things like providers failing to start.
```

Comment:
If this is something we should revisit, I'd mark it with a `TODO` comment so it's greppable.

Reply:
Eh it's less TODO and more "if you come looking at this code because people are hitting lots of bugs in providers failing to start while running convert it's probably ok to weaken the checks here"
But I'd like to understand _why_ that's happening if it does start happening because in general providers should start fine so I'd rather have the bug reports than silent failures.

Reply:
👍🏻 SGTM